components/movementsensor/gpsnmea/gpsnmea.go

```diff
@@ -107,9 +107,10 @@ var model = resource.DefaultModelFamily.WithModel("gps-nmea")
 // NmeaMovementSensor implements a gps that sends nmea messages for movement data.
 type NmeaMovementSensor interface {
 	movementsensor.MovementSensor
-	Start(ctx context.Context) error          // Initialize and run MovementSensor
-	Close(ctx context.Context) error          // Close MovementSensor
-	ReadFix(ctx context.Context) (int, error) // Returns the fix quality of the current MovementSensor measurements
+	Start(ctx context.Context) error                 // Initialize and run MovementSensor
+	Close(ctx context.Context) error                 // Close MovementSensor
+	ReadFix(ctx context.Context) (int, error)        // Returns the fix quality of the current MovementSensor measurements
+	ReadSatsInView(ctx context.Context) (int, error) // Returns the number of satellites in view
```

Comment:
At some point we should move most of these to helper functions on g.data rather than have them as functions on the NmeaMovementSensor interface. I'm not going to block this PR for a refactor that big.